src/browser/BrowserAction.h

```diff
@@ -52,6 +76,7 @@ class BrowserAction
     QJsonObject buildResponse(const QString& action, const QJsonObject& message, const QString& nonce);
     QJsonObject getErrorReply(const QString& action, const int errorCode) const;
     QJsonObject decryptMessage(const QString& message, const QString& nonce);
+    BrowserRequest getBrowserRequest(const QJsonObject& json);
```

Comment:
`toBrowserRequest` or maybe just `decodeRequest`. Parse and Unpack are possible verbs as well. 

Another take on this is that when a request is received, the internals of BrowserService just immediately unpack it into the struct and pass the struct around. Then the only function that takes json is the read function. 

You'd likewise have a response struct as well.

Reply:
The only way to use the struct directly would be to divide the requests in two groups: those which need the decrypted content and association, and those which do not. The first group would handle the struct.